core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -237,6 +240,10 @@ private Path download(Terminal terminal, String pluginId, Path tmpDir) throws Ex
         }
 
         // fall back to plain old URL
+        if (pluginId.contains("://") == false) {
```

Comment:
Actually I reverted this. As I thought, file URLs still must contain double slash (see https://en.wikipedia.org/wiki/File_URI_scheme for example). While java URL will parse a URL with a single slash, it doesn't actually work when trying to read the file.